index.bs

```diff
@@ -384,6 +384,56 @@ scheme, is encoded in the <code>URL</code> field.
   };
 </pre>
 
+<div class="example">
+In this example we build a page supporting a device. To start with when the page
+is loaded it should first check if the device is already plugged in and it has
+permission to access it by calling {{USB/getDevices()}},
+
+<pre>
+  document.addEventListener('DOMContentLoaded', async () => {
+    let devices = await navigator.usb.getDevices();
+    devices.forEach(device => {
+      // Add |device| to the UI.
+    });
+  });
+</pre>
+
+After the page is loaded the user may <a>connect</a> or <a>disconnect</a> a
+device from their system so script should also register for these events in
+order to keep the interface up-to-date,
+
+<pre>
+  navigator.usb.addEventListener('connect', device => {
+    // Add |device| to the UI.
+  });
+
+  navigator.usb.addEventListener('disconnect', device => {
+    // Remove |device| from the UI.
+  });
+</pre>
+
+If this is the first time the user has visited the page then there will be no
+existing permissions so the page must first call {{USB/requestDevice()}} in
+response to a user gesture. In this case the page supports devices from vendor
+<code>0xABCD</code>, that carry the vendor-specific subclass <code>0x01</code>,
+
+<pre>
+  let button = document.getElementById('request-device');
+  button.addEventListener('click', async () => {
+    try {
+      let device = await navigator.usb.requestDevice({ filters: [{
+          vendorId: 0xABCD,
+          classCode: 0xFF, // vendor-specific
+          protocolCode: 0x01
+      }]});
+      // Add |device| to the UI.
```

Comment:
This should probably be after the catch() block, since modifying the UI might cause unrelated exceptions. That'll force you to put a "let device;" above the `try`.